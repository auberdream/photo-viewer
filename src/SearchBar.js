import React, { Component } from 'react';
import Images from './resources/images';

class SearchBar extends Component {
  constructor() {
    super();
    this.state = {
      currentSearch: "",
      foundImages: null
    };
  };

  handleChange = (event) => {
    let searchTerm = event.target.value
    this.setState({
      currentSearch: searchTerm
    }, this.findImages(searchTerm));
  }

  findImages = (searchTerm) => {
    const found = Images.filter(image => image.first_name.includes(searchTerm) ||
                                       image.last_name.includes(searchTerm) ||
                                       image.tag.includes(searchTerm));
    this.setState({
      foundImages: found
    })
  }

  getImages = () => {
    return this.state.foundImages
  }

  isNotEmpty = () => {
    return this.state.currentSearch !== ""
  }

  isCurrentSearch = () => {
    if (this.isNotEmpty()) {
      return
       // TO DO <SearchResults handler={ this.props.handler } currentSearch={ this.getImages() } />
    }
  }

  render() {
    return (
      <div className="searchbar-container">
        <input type="textbox" onChange={ this.handleChange } placeholder="What are you looking for?" />
        { this.isCurrentSearch() }
      </div>
    )
  }
}

export default SearchBar;
