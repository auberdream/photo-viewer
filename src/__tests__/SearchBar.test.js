import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchBar from '../SearchBar';

describe('SearchBar', () => {

  const wrapper = shallow(<SearchBar />)

  it('renders without crashing', () => {
    shallow(<SearchBar />);
  });

  it('renders a text input', () => {
    expect(wrapper.find('input').length).toEqual(1)
  })

});
