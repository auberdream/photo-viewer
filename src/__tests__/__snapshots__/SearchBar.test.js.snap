// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  className="searchbar-container"
>
  <div
    className="fake-textbox"
  >
    <img
      className="magnifying-glass"
      src="./images/magnifying-glass.png"
    />
    <input
      id="textbox"
      onChange={[Function]}
      placeholder="What are you looking for?"
      type="textbox"
    />
  </div>
  <div
    className="searchresults-container"
  />
</div>
`;
